hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/BlockStreamManager.java

```diff
@@ -139,4 +138,9 @@ enum PendingWork {
      * @param timeout the maximum time to wait for block stream shutdown
      */
     void awaitFatalShutdown(@NonNull Duration timeout);
+
+    /**
+     * Resets state associated to node rewards.
+     */
+    void resetNodeRewards();
```

Comment:
Something is very wrong architecturally if `BlockStreamManager` needs to know about node rewards, missing judges and such.

Reply:
The reason this in memory state is maintained in `BlockStreamManager` is because we need to accumulate this every round on `endRound` and write it every block, instead of changing the singleton state every round.

Reply:
@OlegMazurov Changed to use a sub interface of `BlockStreamManager` for  anything other service has to do during endBlock or startBlock.